File: VB/Core/LogViewerVB.Core/ViewModels/LogViewerControlViewModel.vb
Imports MvvmVB.Core

Namespace ViewModels

    Public Class LogViewerControlViewModel : Inherits ViewModel : Implements ILogDataStoreImpl

#Region "Constructors"

        Public Sub New(store As ILogDataStore)
            DataStore = store
        End Sub

#End Region

#Region "Properties"

        Public ReadOnly Property DataStore As ILogDataStore Implements ILogDataStoreImpl.DataStore

#End Region

    End Class

End Namespace

File: CSharp/Core/LogViewer.Core/ViewModels/LogViewerControlViewModel.cs
using Mvvm.Core;

namespace LogViewer.Core.ViewModels;

public class LogViewerControlViewModel : ViewModel, ILogDataStoreImpl
{
    #region Constructor

    public LogViewerControlViewModel(ILogDataStore dataStore)
    {
        DataStore = dataStore;
    }

    #endregion

    #region Properties

    public ILogDataStore DataStore { get; set; }

    #endregion
}